### DISCUSSION TITLE
Alert Group By should also enable a Join to detect missing log records

### DISCUSSION THREAD

**Discussion by mcattle** [datalust/seq-tickets#1865] [Ideas]
When setting up alerts for error messages, using the `group by` feature is a great way to consolidate alerts for several services / jobs into one alert.

However, this does not work for negative detection (`having count = 0`) of expected activity, as the results with no count also have nothing to group by.

For example:
![image](https://github.com/datalust/seq-tickets/assets/7505674/31b6cbe0-8a6b-44c5-8e95-7f5990b8d588)

I'd like to propose extending the query syntax to include something like `select count(*) as count join ['list', 'of', 'job', 'names'] as Job`, so that we can get results for each Job where the count is zero (and then of course include the collection of matching `Job`s in the notification):
```
time                        Job            count
05 June 2023 12:00:00.000   HeartbeatJob   9
05 June 2023 12:00:00.000   OtherJob       0
05 June 2023 15:00:00.000   HeartbeatJob   4
05 June 2023 15:00:00.000   OtherJob       0
05 June 2023 18:00:00.000   HeartbeatJob   2
05 June 2023 18:00:00.000   OtherJob       0
05 June 2023 21:00:00.000   HeartbeatJob   0
05 June 2023 21:00:00.000   OtherJob       0
```

**Comment by nblumhardt** [comment]
Thanks for the suggestion, we're doing some thinking in this area 👍
